ptf/sainat.py

```diff
@@ -0,0 +1,1493 @@
+
+            svi_dnat_pool = sai_thrift_nat_entry_t(
+                vr_id=self.default_vrf, data=svi_nat_data,
+                nat_type=SAI_NAT_TYPE_DESTINATION_NAT_POOL)
+            sai_thrift_create_nat_entry(
+                self.client, svi_dnat_pool, dst_ip=self.svi_nbor_ip,
+                nat_type=SAI_NAT_TYPE_DESTINATION_NAT_POOL)
+
+            sai_thrift_set_router_interface_attribute(
+                self.client, self.ingr_port_rif, nat_zone_id=1)
+            sai_thrift_set_router_interface_attribute(
+                self.client, self.egr_port_rif, nat_zone_id=0)
+
+            sai_thrift_set_router_interface_attribute(
+                self.client, self.ingr_lag_rif, nat_zone_id=1)
+            sai_thrift_set_router_interface_attribute(
+                self.client, self.egr_lag_rif, nat_zone_id=0)
+
+            sai_thrift_set_router_interface_attribute(
+                self.client, self.ingr_svi_rif, nat_zone_id=1)
+            sai_thrift_set_router_interface_attribute(
+                self.client, self.egr_svi_rif, nat_zone_id=0)
+
+            print("\n***Egress L3 port <-***")
+            verify_translation(self.egr_port_rif)
+            print("\n***Egress L3 LAG <-***")
+            verify_translation(self.egr_lag_rif)
+            print("\n***Egress SVI <-***")
+            verify_translation(self.egr_svi_rif)
+
+        finally:
+            sai_thrift_set_router_interface_attribute(
+                self.client, self.ingr_svi_rif, ingress_acl=0)
+            sai_thrift_set_router_interface_attribute(
+                self.client, self.ingr_lag_rif, ingress_acl=0)
+            sai_thrift_set_router_interface_attribute(
+                self.client, self.ingr_port_rif, ingress_acl=0)
+
+            sai_thrift_set_router_interface_attribute(
+                self.client, self.ingr_svi_rif, nat_zone_id=0)
+            sai_thrift_set_router_interface_attribute(
+                self.client, self.egr_svi_rif, nat_zone_id=0)
+
+            sai_thrift_set_router_interface_attribute(
+                self.client, self.ingr_lag_rif, nat_zone_id=0)
+            sai_thrift_set_router_interface_attribute(
+                self.client, self.egr_lag_rif, nat_zone_id=0)
+
+            sai_thrift_set_router_interface_attribute(
+                self.client, self.ingr_port_rif, nat_zone_id=0)
+            sai_thrift_set_router_interface_attribute(
+                self.client, self.egr_port_rif, nat_zone_id=0)
+
+            sai_thrift_remove_nat_entry(self.client, svi_dnat_pool)
+            sai_thrift_remove_nat_entry(self.client, svi_dnat)
+
+            sai_thrift_remove_nat_entry(self.client, lag_dnat_pool)
+            sai_thrift_remove_nat_entry(self.client, lag_dnat)
+
+            sai_thrift_remove_nat_entry(self.client, port_dnat_pool)
+            sai_thrift_remove_nat_entry(self.client, port_dnat)
+
+
+class NatTest(SaiHelper):
+    '''
+    Basic NAT configuration test
+    '''
+
+    def setUp(self):
```

Comment:
It would be helpful to have a diagram here in comments that describes the topology for this test case..

For example:
```
Nbr2 (192.168.0.1/24) - - - - server_port (port24) - DUT-  wan_port (port25) - - - - Nbr1 (20.20.20.1)
00:33:44:55:66:77                  zone=0                       zone=1               00:11:22:33:44:55        
Nhop2                                                                                Nhop1
```

Reply:
done